returnn/extract_prior.py

```diff
@@ -72,54 +78,57 @@ def run(self):
             plt.savefig(self.out_prior_plot.get_path())
 
 
-class ReturnnComputePriorJob(Job):
+class ReturnnComputePriorJob(_ReturnnComputePriorJob):
+    """
+    This Job was deprecated because absolute paths are hashed.
+
+    See https://github.com/rwth-i6/i6_core/issues/306 for more details.
+    """
+
+    pass
+
+
+class ReturnnComputePriorJobV2(Job):
     """
     Given a model checkpoint, run compute_prior task with RETURNN
     """
 
     def __init__(
         self,
-        model_checkpoint,
-        returnn_config,
-        prior_data=None,
+        model_checkpoint: Checkpoint,
+        returnn_config: ReturnnConfig,
+        prior_data: Optional[Dict[str, Any]] = None,
         *,
-        log_verbosity=3,
-        device="gpu",
-        time_rqmt=4,
-        mem_rqmt=4,
-        cpu_rqmt=2,
-        returnn_python_exe=None,
-        returnn_root=None,
+        log_verbosity: int = 3,
+        device: str = "gpu",
+        time_rqmt: float = 4,
+        mem_rqmt: float = 4,
+        cpu_rqmt: int = 2,
+        returnn_python_exe: Optional[tk.Path] = None,
+        returnn_root: Optional[tk.Path] = None,
     ):
         """
-        :param Checkpoint model_checkpoint:  TF model checkpoint. see `ReturnnTrainingJob`.
-        :param ReturnnConfig returnn_config: object representing RETURNN config
-        :param dict[str]|None prior_data: dataset used to compute prior (None = use one train epoch)
-        :param int log_verbosity: RETURNN log verbosity
-        :param str device: RETURNN device, cpu or gpu
-        :param float|int time_rqmt: job time requirement in hours
-        :param float|int mem_rqmt: job memory requirement in GB
-        :param float|int cpu_rqmt: job cpu requirement in GB
-        :param tk.Path|str|None returnn_python_exe: path to the RETURNN executable (python binary or launch script)
-        :param tk.Path|str|None returnn_root: path to the RETURNN src folder
+        :param model_checkpoint:  TF model checkpoint. see `ReturnnTrainingJob`.
+        :param returnn_config: object representing RETURNN config
+        :param prior_data: dataset used to compute prior (None = use one train epoch)
+        :param log_verbosity: RETURNN log verbosity
+        :param device: RETURNN device, cpu or gpu
+        :param time_rqmt: job time requirement in hours
+        :param mem_rqmt: job memory requirement in GB
+        :param cpu_rqmt: job cpu requirement in GB
+        :param returnn_python_exe: path to the RETURNN executable (python binary or launch script)
+        :param returnn_root: path to the RETURNN src folder
         """
         assert isinstance(returnn_config, ReturnnConfig)
         kwargs = locals()
         del kwargs["self"]
 
         self.model_checkpoint = model_checkpoint
 
-        self.returnn_python_exe = (
```

Comment:
If we remove this, we also need to remove the default `None`, otherwise the job will crash.

Reply:
if we incorporate https://github.com/rwth-i6/i6_core/pull/261 then we could leave default `None` which then falls back to the system python (or some virtual env that is used to run the manager)